native/cocos/renderer/gfx-vulkan/VKBuffer.cpp

```diff
@@ -42,86 +42,83 @@ CCVKBuffer::~CCVKBuffer() {
 
 void CCVKBuffer::doInit(const BufferInfo & /*info*/) {
     _gpuBuffer = ccnew CCVKGPUBuffer;
-    _gpuBuffer->usage = _usage;
-    _gpuBuffer->memUsage = _memUsage;
     _gpuBuffer->size = _size;
-    _gpuBuffer->stride = _stride;
     _gpuBuffer->count = _count;
-
-    if (hasFlag(_usage, BufferUsageBit::INDIRECT)) {
-        const size_t drawInfoCount = _size / sizeof(DrawInfo);
-        _gpuBuffer->indexedIndirectCmds.resize(drawInfoCount);
-        _gpuBuffer->indirectCmds.resize(drawInfoCount);
-    }
-
-    cmdFuncCCVKCreateBuffer(CCVKDevice::getInstance(), _gpuBuffer);
-    CCVKDevice::getInstance()->getMemoryStatus().bufferSize += _size;
-    CC_PROFILE_MEMORY_INC(Buffer, _size);
+    createBuffer();
 
     _gpuBufferView = ccnew CCVKGPUBufferView;
+    _gpuBufferView->range = _size;
     createBufferView();
 }
 
 void CCVKBuffer::doInit(const BufferViewInfo &info) {
     auto *buffer = static_cast<CCVKBuffer *>(info.buffer);
     _gpuBuffer = buffer->gpuBuffer();
     _gpuBufferView = ccnew CCVKGPUBufferView;
+    _gpuBufferView->range = _size;
     createBufferView();
 }
 
+void CCVKBuffer::createBuffer() {
```

Comment:
It seems this function just initialize _gpuBuffer, not create it. So the function name is not good.